xbmc/interfaces/json-rpc/FileItemHandler.cpp

```diff
@@ -43,6 +43,31 @@ using namespace MUSIC_INFO;
 using namespace JSONRPC;
 using namespace XFILE;
 
+namespace
+{
+/**
+   * @brief Append the provided listitem properties to the item
+   *
+   * @param parameterObject the parameters passed on the request. Must include a "properties" array of objects
+   *  (e.g. "properties": [{"key": "inputstream", "value": "inputstream.adaptive"}, ...})
+   * @param item the listitem to append properties
+   */
+void AppendUserProvidedListItemProperties(const CVariant& parameterObject, const CFileItemPtr& item)
+{
+  if (parameterObject.isMember("properties") && parameterObject["properties"].isArray())
+  {
+    for (CVariant::const_iterator_array field = parameterObject["properties"].begin_array();
```

Comment:
from a build test (and just as side note): you may want to use 
```c
for (auto field = parameterObject["properties"].begin_array();
```

what does not work is `const auto field =` because then it's not possible to increment the iterator `field++`.
not sure why it is possible to increment when using explicit `CVariant::const_iterator_array`.. is that a design flaw in CVariant or am i misunderstanding?